Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -0,0 +1,1032 @@
+    fNameChannel[imod][ich] = namech;
+     std::vector<int> coord;
+     coord.push_back (imod);
+     coord.push_back (ich);
+     fMapChNameModCh.insert(std::pair<std::string,std::vector<int>>(namech,coord));
+}
+
+bool ZDCRawDataTask::getModAndCh(std::string chName,int * module, int * channel){
+    if (chName.compare("NONE")== 0){
+        return true;
+    }
+    if (fMapChNameModCh.find(chName) != fMapChNameModCh.end()) {
+        *module=fMapChNameModCh[chName].at(0);
+        *channel=fMapChNameModCh[chName].at(1);
+    return true;
+    }
+    return false;
+}
+
+void ZDCRawDataTask::setBinHisto1D(int numBinX, double minBinX, double maxBinX){
+    setNumBinX(numBinX);
+    setMinBinX(minBinX);
+    setMaxBinX(maxBinX);
+}
+
+void ZDCRawDataTask::setBinHisto2D(int numBinX, double minBinX, double maxBinX, int numBinY, double minBinY, double maxBinY){
+    setNumBinX(numBinX);
+    setMinBinX(minBinX);
+    setMaxBinX(maxBinX);
+    setNumBinY(numBinY);
+    setMinBinY(minBinY);
+    setMaxBinY(maxBinY);
+}
+
+
+bool ZDCRawDataTask::addNewHisto(std::string type,std::string name,std::string title,std::string chName, std::string condition){
+    int mod;
+    int ch;
+    int ih; // index Histogram
+    infoHisto1D h1d;
+    infoHisto2D h2d;
+    // Check if the channel name defined
+    if (getModAndCh(chName,&mod, &ch) == true){
+    // Check if channel selected produced data. (_SP spare _OTR only trigger)
+        if (chName.find("_SP") != std::string::npos || chName.find("_OTR") != std::string::npos)
+            return false;
+        TString hname = TString::Format("%s",name.c_str());
+        TString htit = TString::Format("%s",title.c_str());
+//BASELINE
+        if ( type.compare("BASELINE")== 0){
+    // Check if Histogram Exist
+            if (std::find(fNameHisto.begin(), fNameHisto.end(), name) == fNameHisto.end()) {
+                fNameHisto.push_back(name);
+                ih = fMapBinNameIdSummaryHisto.size();
+                fMapBinNameIdSummaryHisto.insert(std::pair<std::string,int>(chName,ih));
+                h1d.histo=new TH1F(hname, htit, fNumBinX, fMinBinX,fMaxBinX);
+                h1d.condHisto.push_back(condition);
+                ih = (int)fMatrixHistoBaseline[mod][ch].size();
+                fMatrixHistoBaseline[mod][ch].push_back(h1d);
+
+                if (ih < (int)fMatrixHistoBaseline[mod][ch].size()){
+                    getObjectsManager()->startPublishing(fMatrixHistoBaseline[mod][ch].at(ih).histo);
+                    try {
+                        getObjectsManager()->addMetadata(fMatrixHistoBaseline[mod][ch].at(ih).histo->GetName(), "custom", "34");
+                        return true;
+                    }
+                    catch (...) {
+                        ILOG(Warning, Support) << "Metadata could not be added to " << fMatrixHistoBaseline[mod][ch].at(ih).histo->GetName() << ENDM;
+                        return false;
+                    }
```

Comment:
you probably do not need this, it was an example
```suggestion
```

Reply:
Ok I'll remove it.